lib/ansible/modules/file.py

```diff
@@ -552,35 +552,40 @@ def execute_touch(path, follow, timestamps):
     result = {'dest': path}
     mtime = get_timestamp_for_time(timestamps['modification_time'], timestamps['modification_time_format'])
     atime = get_timestamp_for_time(timestamps['access_time'], timestamps['access_time_format'])
+    diff = initial_diff(path, 'touch', prev_state)
+    file_args = module.load_file_common_arguments(module.params)
 
-    if not module.check_mode:
-        if prev_state == 'absent':
-            # Create an empty file if the filename did not already exist
+    # If the file did not already exist
+    if prev_state == 'absent':
+        # if we are in check mode and the file is absent
+        # we can set the changed status to True and return
+        if module.check_mode:
+            result['changed'] = True
+            return result
+        else:
```

Comment:
Since the `if` returns early, you can drop the `else` and unindent creating the empty file.

Reply:
Oh correct, it is now not needed anymore